### PR TITLE
fix(op_crates/web): Strip username and password from location

### PR DESCRIPTION
**Pull request by nayeemrmn**

### CODE REVIEW

File: op_crates/web/12_location.js [hunk 1]
```diff
@@ -10,6 +10,9 @@
         throw new TypeError("Illegal constructor.");
       }
       const url = new URL(href);
+      url.username = "";
+      url.password = "";
+      console.trace(url.href);
```
Comment (lucacasonato) [inline op_crates/web/12_location.js:15]: stray debug statement here :-)

Comment (nayeemrmn) [reply op_crates/web/12_location.js:15]:
```suggestion
```

**Comment by bartlomieju** [comment]
Maybe we could use same code that was removed in https://github.com/denoland/deno/pull/5034?

**Comment by nayeemrmn** [comment]
> Maybe we could use same code for `location` that was removed in #5034?

What's the significance of that implementation? The differences from that one are purposeful.

**Comment by bartlomieju** [comment]
> > Maybe we could use same code for `location` that was removed in #5034?
> 
> What's the significance of that implementation? The differences from that one are purposeful.

There's none; I just thought it might be useful.

**Comment by nayeemrmn** [comment]
Closing in favour of #9084.

**Pull request closed without merging**